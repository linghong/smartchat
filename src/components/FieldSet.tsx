import { FC, ReactNode } from 'react';

interface FieldSetProp {
  children: ReactNode;
}

const FieldSet: FC<FieldSetProp> = ({ children }) => {
  return (
    <div className="flex flex-col my-3 py-4 px-6 bg-slate-50 border border-indigo-100 shadow-md rounded">
      {children}
    </div>
  );
};
export default FieldSet;
