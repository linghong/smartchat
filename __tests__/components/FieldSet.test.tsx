
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import FieldSet from '@/src/components/FieldSet'

describe('FieldSet Component', () => {
  test('renders children correctly', () => {
    const { getByText } = render(
      <FieldSet>
        <p>Test Child</p>
      </FieldSet>
    )
    
    expect(getByText('Test Child')).toBeInTheDocument()
  })

  test('applies correct classes', () => {
    const { container } = render(
      <FieldSet>
        <p>Test Child</p>
      </FieldSet>
    );

    expect(container.firstChild).toHaveClass('flex')
    expect(container.firstChild).toHaveClass('flex-col')
    expect(container.firstChild).toHaveClass('bg-slate-50')
    expect(container.firstChild).toHaveClass('shadow-md')
    expect(container.firstChild).toHaveClass('rounded')
  });

  test('matches the snapshot', () => {
    const { asFragment } = render(
      <FieldSet>
        <p>Test Child</p>
      </FieldSet>
    );

    expect(asFragment()).toMatchSnapshot()
  })
})