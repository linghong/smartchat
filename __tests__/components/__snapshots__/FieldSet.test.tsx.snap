// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldSet Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col my-3 py-4 px-6 bg-slate-50 border border-indigo-100 shadow-md rounded"
  >
    <p>
      Test Child
    </p>
  </div>
</DocumentFragment>
`;
